src/main/java/org/testcontainers/vault/VaultContainer.java

```diff
@@ -0,0 +1,132 @@
+package org.testcontainers.vault;
+
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.traits.LinkableContainer;
+import org.testcontainers.shaded.com.github.dockerjava.api.command.InspectContainerResponse;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.testcontainers.shaded.com.github.dockerjava.api.model.Capability.IPC_LOCK;
+
+/**
+ * GenericContainer subclass for Vault specific configuration and features. The main feature is the
+ * withSecretInVault method, where users can specify which secrets to be pre-loaded into Vault for
+ * their specific test scenario.
+ *
+ * Other helpful features include the withVaultPort, and withVaultToken methods for convenience.
+ */
+public class VaultContainer<SELF extends VaultContainer<SELF>> extends GenericContainer<SELF>
+        implements LinkableContainer {
+
+    private static final String VAULT_PORT="8200";
```

Comment:
Please surround binary operators with two spaces (I assume it's a refactoring error here ;)).